feat: add git pull before opening VSCode in runTask

roomote opened the pull request:
## Summary

This PR adds a Already up to date. operation before opening VSCode in the  function to ensure that we always have the latest version of the repository we're working in.

## Changes

- **Added repo configuration object** ():
  - Contains configuration for available repos:  and 
  - Provides helper functions to get repo config by path or name

- **Modified  function** ():
  - Added Already up to date. operation before opening VSCode
  - Uses repo configuration to determine the correct path for git pull
  - Includes fallback to workspace directory if repo config not found
  - Continues execution even if git pull fails to avoid blocking task execution
  - Added proper logging for git pull operations

- **Added comprehensive tests** ():
  - Tests for repo configuration functionality
  - Validates correct repo paths and descriptions
  - Tests helper functions for finding repos by path and name

## Benefits

- Ensures tasks always run against the latest code
- Reduces issues caused by outdated repository state
- Maintains backward compatibility with existing workflows
- Graceful handling of git pull failures

## Testing

- ✅ All new tests pass
- ✅ Existing functionality unchanged
- ✅ Linting and type checking pass
- ✅ Graceful error handling for git pull failures
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds git pull operation in `runTask` to ensure latest repo version, with repo configuration and tests.
> 
>   - **Behavior**:
>     - Adds git pull operation in `runTask` in `runTask.ts` to ensure latest repo version before opening VSCode.
>     - Uses `getRepoConfig` to determine repo path for git pull, with fallback to all repos if not found.
>     - Continues execution even if git pull fails, with logging for success and failure.
>   - **Repo Configuration**:
>     - Adds `repoConfig.ts` with `RepoConfig` interface and `REPO_CONFIGS` array for repo details.
>     - Provides `getRepoConfig` and `getRepoConfigByName` functions to retrieve repo configurations.
>   - **Testing**:
>     - Adds `repoConfig.spec.ts` to test repo configuration retrieval by path and name, and validate repo details.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 497ee4e2d76ad146afaa4bb951fd3c1fcf861fbd. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

cte commented on packages/evals/src/cli/runTask.ts:
```diff
@@ -156,6 +157,29 @@ export const runTask = async ({ run, task, publish, logger }: RunTaskOptions) =>
 	const containerized = isDockerContainer()
 
+	// Perform git pull to ensure we have the latest version of the repo
+	const repoConfig = getRepoConfig(workspacePath)
+	if (repoConfig) {
+		logger.info(`performing git pull for repo: ${repoConfig.name} at ${repoConfig.path}`)
+		try {
+			const gitPullResult = await execa("git", ["pull"], { cwd: repoConfig.path })
+			logger.info(`git pull completed successfully: ${gitPullResult.stdout}`)
+		} catch (error) {
+			logger.warn(`git pull failed for ${repoConfig.name}: ${error}`)
+			// Continue execution even if git pull fails
+		}
+	} else {
+		// Fallback: try git pull in the workspace directory
+		logger.info(`performing git pull in workspace: ${workspacePath}`)
+		try {
+			const gitPullResult = await execa("git", ["pull"], { cwd: workspacePath })
+			logger.info(`git pull completed successfully: ${gitPullResult.stdout}`)
+		} catch (error) {
+			logger.warn(`git pull failed in workspace ${workspacePath}: ${error}`)
+			// Continue execution even if git pull fails
+		}
+	}
```
In if the `workspacePath` doesn't exactly match a repo path then we should run `git pull` on all repos.

cte replied:
@roomote In if the `workspacePath` doesn't exactly match a repo path then we should run `git pull` on all repos.

roomote commented:
✅ **Implementation Complete**

I have successfully implemented the requested change. When the `workspacePath` does not exactly match a repo path, the system now runs `git pull` on all configured repositories instead of just the workspace directory.

**Changes Made:**
- Modified the `runTask` function in `packages/evals/src/cli/runTask.ts`
- When no exact repo match is found, the system now iterates through all `REPO_CONFIGS` and runs `git pull` on each repository
- Enhanced logging to clearly indicate when pulling all repos vs a single repo
- All existing tests pass and TypeScript compilation is successful

**Commit:** 497ee4e2d - "feat: run git pull on all repos when workspacePath does not match"

This ensures that all repositories are kept up to date regardless of the workspace location, addressing your feedback perfectly. 🚀
